File: creating-non-existing-dates/erlang/impossible_dates.erl
-module(impossible_dates).

-export([examples/0]).

examples() ->
    %% Erlang valid date as tuple
    ValidDate = {2021, 1, 15},

    io:format("~p~n", [ValidDate]), % prints {2021,1,15}

    %% Erlang invalid date as tuple
    InvalidDate = {2021, 2, 30},

    io:format("~p~n", [InvalidDate]). % prints {2021,2,30}

File: creating-non-existing-dates/elixir/lib/impossible_dates.ex
defmodule ImpossibleDates do
  def examples do
    # Elixir Date.new/3 with valid input
    valid_date = Date.new(2021, 1, 15)

    IO.inspect(valid_date) # prints {:ok, ~D[2021-01-15]}

    # Elixir Date.new/3 with invalid input
    invalid_date = Date.new(2021, 2, 30)

    IO.inspect(invalid_date) # prints {:error, :invalid_date}
  end
end
